igvc_perception/src/train_eval/IGVCDataset.py

```diff
@@ -49,11 +50,26 @@ def __getitem__(self, index):
             img = cv2.imread(img_path)
         except:
             pdb.set_trace()
-        target = cv2.imread(target_path)[:,:,2]
-        
+
+        if self.preprocessor is not None:
+            img = self.preprocessor(img)
+
+        target = cv2.imread(target_path)
+
+        # randomly flip the image vertically sometimes
+        if (np.random.randint(2) == 1):
+            img = cv2.flip(img,1)
+            target = cv2.flip(target,1)
+
+        #Access to red color only (OpenCV: BGR) This is because the lines are categorized as red color
+        # And we are interested in lines for the line detection.
+        target = target[:,:,2]
         img = cv2.resize(img, (self.im_size[1],self.im_size[2]))
         target = cv2.resize(target, (self.im_size[1], self.im_size[2]))
         target[target != 0] = 255
+        #cv2.imshow("original image", img)
+        #cv2.imshow("target", target) #Check if the image is collectlly displayed as a binary image
```

Comment:
I think you should uncomment these lines, delete them, or add some kind of logic to thoroughly ensure that it is displayed as a binary image.

Reply:
Another possibility to just space it out and better distinguish it as a checking mechanism.

Reply:
Thanks @suhacker1, I updated my code based on your feedback!